notebook/services/contents/tests/test_manager.py

```diff
@@ -129,6 +129,26 @@ def test_bad_symlink(self):
             # broken symlinks should still be shown in the contents manager
             self.assertTrue('bad symlink' in contents)
 
+    @dec.skipif(sys.platform == 'win32' and sys.version_info[0] < 3)
+    def test_recursive_symlink(self):
+        with TemporaryDirectory() as td:
+            cm = FileContentsManager(root_dir=td)
+            path = 'test recursive symlink'
+            _make_dir(cm, path)
+            os_path = cm._get_os_path(path)
+            os.symlink("recursive", os.path.join(os_path, "recursive"))
+            file_model = cm.new_untitled(path=path, ext='.txt')
+
+            model = cm.get(path)
+
+            contents = {
+                content['name']: content for content in model['content']
+            }
+            self.assertTrue('untitled.txt' in contents)
+            self.assertEqual(contents['untitled.txt'], file_model)
+            # recusrive symlinks should not be shown in the contents manager
+            self.assertFalse('recusrive' in contents)
```

Comment:
Spelling of 'recursive'.

Also, we should use the `assertIn` and `assertNotIn` methods rather than assertTrue and assertFalse. The more specific methods give more useful messages when things fail.